specs/AppLifecycle/Activation/AppLifecycle Activation.md

```diff
@@ -0,0 +1,245 @@
+- Which activation kinds will unpackaged apps be able to use?
+- How does the app tell the system which activation kinds it wants to use?
+
+
+### Supported activation kinds
+
+The first release focuses on the most commonly-used rich activation kinds. The AppLifecycle component will provide a converged GetActivatedEventArgs API which will get all args, regardless of activation kind, effectively merging in the functionality of GetCommandLineArgs/GetCommandLineW. It will also be decoupled from multi-instancing.
+
+Of the 44 UWP activation kinds, 5 of the most commonly-used kinds are also supported for Desktop Bridge and Windows [reg-free WinRT](https://blogs.windows.com/windowsdeveloper/2019/04/30/enhancing-non-packaged-desktop-apps-using-windows-runtime-components/) apps. The AppLifecycle implementation is based on this set. Some activation kinds require prior registration, others do not. The target list for v1 is as follows:
+
+|     Activation   kind    |     Description                                                                                                                                             |
+|--------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------|
+|     Launch               |     Launch   from a Start tile/shortcut, from a command window, or programmatically via   ShellExecute/CreateProcess.                                       |
+|     File                 |     The   app is activated when the user double-clicks a registered file, or when a   caller calls ShellExecute or LaunchFileAsync on a registered file.    |
+|     Protocol             |     Activated   via ShellExecute or LaunchUriAsync, or by specifying a protocol string on a   command-line.                                                 |
+|     StartupTask          |     The   app starts on user log-in - either registered in the registry, or via   shortcuts in a well-known startup folder.                                 |
+|     ToastNotification    |     The   app sends a toast, the user responds, and this activates the app.                                                                                 |
+    
+
+### Activation registration
+Apps can start using supported modern activation phases by:
+- Calling a method to register activation.
+- Calling a method in their entrypoint to handle activation.
+
+From a Win32 app perspective, the 2 activation kinds that do not require any registration (Shortcut/Tile Launch and CommandLine) are effectively the same in terms of activation payload. Indeed, it is not possible to distinguish a launch activation of a Win32 app from a commandline activation. Therefore, the AppLifecycle implementation will merge Launch and CommandLine into one (Launch) activation kind. 
```

Comment:
the shortcut file that launches the app can be retrieved and can be used to provide context for the launch. for example the console uses the shortcut to store configuration data, colors, window size, etc. given the property set stored in the shortcut this can be a general purpose configuration store.

consider providing access to the shortcut that launched the app, or its property store.

Reply:
Is there an API for this already, or something we'd add as part of this effort? A little naive bing-ing doesn't lead me anywhere useful.